package/MDAnalysis/topology/ITPParser.py

```diff
@@ -575,13 +583,21 @@ def parse(self, include_dir='/usr/local/gromacs/share/gromacs/top/',
 
         if not all(self.masses):
             empty = self.masses == ''
-            self.masses[empty] = guessers.guess_masses(
-                guessers.guess_types(self.types)[empty])
-            attrs.append(Masses(np.array(self.masses, dtype=np.float64),
-                                guessed=True))
+            self.masses[empty] = Masses.missing_value_label
+
+        attrs.append(Masses(np.array(self.masses, dtype=np.float64),
+                            guessed=False))
+
+        self.elements = DefaultGuesser(None).guess_types(self.names)
+        if all(e.capitalize() in SYMB2Z for e in self.elements):
+            attrs.append(Elements(np.array(self.elements,
+                         dtype=object), guessed=True))
+
```

Comment:
Hm, I thought the approach for most other parsers was to fall back on `Universe.guess_TopologyAttributes` instead of guessing inside the parser itself? I don't think `Elements` is currently guessed inside the ITPParser, is this new addition required for anything?

Reply:
This parser used to read types but partially guess masses from guessed type values instead of the read ones. So, to keep this behavior I had to add a guessed element attribute to this parser so that when partial mass guessing takes place later, element values will be the reference to guess masses from